### DISCUSSION TITLE
Could we get a Processor Directive added to the plugin?

### DISCUSSION THREAD

**Discussion by JamesAiken** [yasirkula/UnityNativeGallery#173] [Ideas]
In NativeGallery.cs at the very top of the file is...

`#if !UNITY_EDITOR && ( UNITY_ANDROID || UNITY_IOS )
using NativeGalleryNamespace;
#endif`

I think this statement should include 
`#define NATIVE_GALLERY`

The reason I ask is because the way I'm using the plugin is through a wrapper class I wrote. This way accessing the Native Gallery functionality stays within one place, and making any updates to one script is easier than updating multiple scripts if I were to use Native Gallery in more than one. With the code being in one place, the directive allows me to wrap up some code that would only run if its defined. 

I can also just define the directive myself, but anytime I would get a new update for the plugin I'd have to redefine it.

**Comment by yasirkula** [comment]
NativeGalleryNamespace contains Android & iOS wrapper classes that get compiled only when building to those platforms. Hence there is that `#if` directive. There isn't a directive that says whether or not NativeGallery is imported. I'm currently not planning to add such a directive but thank you for the feedback!
